src/rgw/rgw_sal_motr.cc

```diff
@@ -2225,64 +2262,36 @@ int MotrObject::get_bucket_dir_ent(const DoutPrefixProvider *dpp, rgw_bucket_dir
         return rc;
       }
 
-      iter = bl.cbegin();
+      bufferlist& blr = bl;
+      iter = blr.cbegin();
       ent.decode(iter);
-      rc = 0;
       // Put into the cache
       this->store->get_obj_meta_cache()->put(dpp, this->get_name(), bl);
```

Comment:
use to_str() for putting specific version of object into the cache.

Reply:
done